hs_geographic_feature_resource/receivers.py

```diff
@@ -0,0 +1,522 @@
+__author__ = 'drew'
+
+
+from django.dispatch import receiver
+from hs_core.signals import *
+from hs_geographic_feature_resource.models import GeographicFeatureResource, OriginalCoverage, GeographicFeatureMetaData
+from hs_core import hydroshare
+import tempfile
+import shutil
+import os
+import pprint
+from hs_geographic_feature_resource.parse_lib import *
+from hs_geographic_feature_resource.forms import *
+import zipfile, json
+from hs_core.hydroshare.resource import ResourceFile
+
+def is_shapefiles(files):
+#check if uploaded files are valid shapefiles (shp, shx, dbf)
+    fn_list=[]
+    for file in files:
+        fn_list.append(file.name)
+    return check_fn_for_shp(fn_list)
+
+
+def check_fn_for_shp(files):
+#used by is_shapefiles
+#check a list of filenames contains valid shapefiles (shp, shx, dbf)
+    shp, shx, dbf = False, False, False
+    all_have_same_filename = False
+    shp_filename, shx_filename, dbf_filename =None, None, None
+    dir_count=0 # can have 0 or 1 folder
+    if len(files) >= 3: # at least have 3 files: shp, shx, dbf
+        for file in files:
+            if file[len(file)-1]=='/':
+                dir_count+=1
+            else:
+                fileName, fileExtension = os.path.splitext(file)
+                if ".shp" == fileExtension:
+                    shp_filename = fileName
+                    shp=True
+                elif ".shx" == fileExtension:
+                    shx_filename = fileName
+                    shx=True
+                elif ".dbf" == fileExtension:
+                    dbf_filename = fileName
+                    dbf=True
```

Comment:
Does any of this code handle the condition where two of the same extension are provided?